.github/workflows/build-test-ship.yaml

```diff
@@ -183,34 +190,251 @@ jobs:
             ${{ env.ECR_REGISTRY }}/${{ env.ECR_REPOSITORY }}:${{ env.IMAGE_TAG }}
             ${{ env.ECR_REGISTRY }}/${{ env.ECR_REPOSITORY }}:latest
 
+  build-for-target-platforms:
+    name: Verify ${{ matrix.service }} images can be built on target platforms
+    runs-on: ubuntu-latest
+    strategy:
+      fail-fast: false
+      matrix:
+        service: ${{ fromJson(inputs.services) }}
+    if: |
+      (github.event_name == 'pull_request' &&
+        !github.event.pull_request.draft) ||
+      (github.event_name == 'push' &&
+        steps.check-version-change.outputs.appchange == 'false')
+    steps:
+      - name: Checkout repo
+        uses: actions/checkout@v3
+
+      - name: Set up QEMU
+        uses: docker/setup-qemu-action@v1
+
+      - name: Set up Docker Buildx
+        uses: docker/setup-buildx-action@v1
+
+      - name: Build images on target platforms
+        uses: docker/build-push-action@v2
+        with:
+          context: .
+          file: ./cmd/${{ matrix.service }}/Dockerfile
+          cache-from: type=gha
+          cache-to: type=gha,mode=max
+          platforms: linux/amd64,linux/arm64
+
+  run-api-proctor-tests:
+    name: Run API Proctor tests
+    runs-on: ubuntu-latest
+    needs: build-and-ship-images
+    timeout-minutes: ${{ inputs.api-proctor-timeout-minutes }}
+    env:
+      PGADMIN_PORT: 5000
+      POSTGRES_USER: postgres
+      POSTGRES_PASSWORD: postgres
+      POSTGRES_SERVER: postgres
+      POSTGRES_PORT: 5432
+    steps:
+      - name: Checkout development repo
+        uses: actions/checkout@v3
+        with:
+          repository: nicheinc/development
+          token: ${{ secrets.NPM_WRITE_TOKEN }}
+          submodules: true
+
+      - name: Download docker image artifacts
+        uses: actions/download-artifact@v3
+
+      - name: Set up yq for YAML processing
+        uses: mikefarah/yq@v4.27.3
+
+      - name: Get relevant services as a space-separated string
+        id: get-services
+        shell: bash
+        run: |
+          tools=`yq -o=json '.services
+            | with_entries(select(.value.profiles.[] | contains("tools")))
+            | to_entries
+            | map(.key)' docker-compose.yml`
+          all_services=`yq -o=json '.services | keys' docker-compose.yml`
+          non_tools=`jq --null-input "${all_services} - ${tools}"`
+          input_services='${{ inputs.services }}'
+          services=`jq --null-input "${input_services} - (${input_services} - ${non_tools})"`
+          echo "services=`jq -r 'join(" ")' <<< ${services}`" >> $GITHUB_OUTPUT
```

Comment:
This is functional but horrible because I'm bad at `yq` and `jq`. Suggestions for simplifying this step appreciated!